### ISSUE TITLE
systemdlint finds errors but doesn't return an error exit code 

### ISSUE THREAD

**Issue by Joe-Heffer-Shef** [priv-kweihmann/systemdlint#73]
Hi, I'm running `systemdlint` as part of a GitHub Actions workflow. Unfortunately, it doesn't cause an error to be detected even when errors are found, so the automatic check appears to pass.

For example:

```
Run systemdlint  ./systemd/*
my-service.service:1:error [Security.Delegate] - Service should have Delegate being set
my-service.service:9:error [ExecNotFound] - Command referenced not found
```

This is the workflow file:

```yaml
name: Lint systemd units
on: [ push ]
jobs:
  lint:
    runs-on: ubuntu-24.04
    steps:
      - name: Checkout
        uses: actions/checkout@v4.2.2
      - name: Install systemdlint
        run: pip install systemdlint
      - name: Lint systemd units
        run: systemdlint  ./systemd/*
```

**Comment by priv-kweihmann** [comment]
Merged
